ale_linters/perl/languageserver.vim

```diff
@@ -0,0 +1,40 @@
+" Authors: ophirr33 <coghlan.ty@gmail.com> and rymdbar
+" Description: Perl::LanguageServer for perl, from cpan.org
+
+" This should have the same value as in perl.vim
+call ale#Set('perl_perl_executable', 'perl')
+" Please note that perl_perl_options does not exist here.
+
+function! ale_linters#perl#languageserver#GetProjectRoot(buffer) abort
+    " Makefile.PL, https://perldoc.perl.org/ExtUtils::MakeMaker
+    " Build.PL, https://metacpan.org/pod/Module::Build
+    " dist.ini, https://metacpan.org/pod/Dist::Zilla
+    let l:potential_roots = [ 'Makefile.PL', 'Build.PL', 'dist.ini', '.git' ]
+
+    for l:root in l:potential_roots
+        let l:project_root = ale#path#FindNearestFile(a:buffer, l:root)
+
+        if empty(l:project_root)
+            let l:project_root = ale#path#FindNearestFile(
+            \   a:buffer,
+            \   l:root,
+            \ )
```

Comment:
This looks like a bug. You're calling the same function twice if the result is empty, and the result will again be empty here.